src/mono/msbuild/android/build/AndroidBuild.targets

```diff
@@ -41,40 +41,29 @@
       <_MonoHeaderPath>$([MSBuild]::NormalizeDirectory($(MicrosoftNetCoreAppRuntimePackRidNativeDir), 'include', 'mono-2.0'))</_MonoHeaderPath>
     </PropertyGroup>
 
-    <!-- Make sure marshal-ilgen is included in the components list. -->
-    <ItemGroup Condition="'$(RuntimeComponents)' != '*'">
-      <_RuntimeComponentList Include="$(RuntimeComponents)" />
-      <_RuntimeComponentList Include="marshal-ilgen" KeepDuplicates="false"/>
+    <ItemGroup Condition="'$(UseNativeAOTRuntime)' != 'true'">
+      <RuntimeComponents Condition="'$(UseAllRuntimeComponents)' == 'true'" Include="@(_MonoRuntimeAvailableComponents)"/>
+      <!-- Make sure marshal-ilgen is included in the components list. -->
+      <RuntimeComponents Condition="'$(UseAllRuntimeComponents)' != 'true'" Include="marshal-ilgen" KeepDuplicates="false"/>
     </ItemGroup>
 
-    <PropertyGroup Condition="'$(RuntimeComponents)' != '*'">
-      <RuntimeComponents>@(_RuntimeComponentList)</RuntimeComponents>
-    </PropertyGroup>
-
     <ItemGroup Condition="'$(_IsLibraryMode)' == 'true'">
       <_CommonLinkerArgs Include="-l:libz.so" />
       <_CommonLinkerArgs Include="-l:liblog.so" />
       <_CommonLinkerArgs Include="-l:libc.so" />
       <_CommonLinkerArgs Include="-l:libm.so" />
       <_CommonLinkerArgs Include="--build-id=sha1" />
 
-      <!-- add all non stub libs first -->
-      <!-- if RuntimeComponents is empty, exclude -static.a and include -stub-static.a instead -->
-      <!-- if RuntimeComponents is *, we're ok because all -static.a is included -->
-      <!-- if RuntimeComponents is a list, add to items and only pull in -static.a -->
-
-      <_UsedComponents 
-        Condition="'$(RuntimeComponents)' != '' and '$(RuntimeComponents)' != '*'"
-        Include="$(RuntimeComponents)" />
+      <!-- include all libraries except components -->
+      <!-- include all component stub libraries -->
+      <!-- if RuntimeComponents is empty, we're ok because we included all the stubs already -->
+      <!-- if RuntimeComponents is not empty, remove stubs for the required components and include the actual component library -->
 
-      <_RuntimeLibraries 
-        Include="$(AndroidBuildDir)\*-stub-static.a" />
-      <_RuntimeLibraries 
-        Include="$(AndroidBuildDir)\*.a"
-        Exclude="$(AndroidBuildDir)\*-static.a" />
+      <_RuntimeLibraries Include="$(AndroidBuildDir)\*.a" Exclude="$(AndroidBuildDir)\libmono-component-*.a" />
+      <_RuntimeLibraries Include="$(AndroidBuildDir)\libmono-component-*-stub-static.a" />
+      <_RuntimeLibraries Condition="'@(RuntimeComponents)' != ''" Remove="$(AndroidBuildDir)\libmono-component-%(RuntimeComponents.Identity)-stub-static.a" />
+      <_RuntimeLibraries Condition="'@(RuntimeComponents)' != ''" Include="$(AndroidBuildDir)\libmono-component-%(RuntimeComponents.Identity)-static.a" />
```

Comment:
The comments don't quite correspond to the actual code.

```xml
      <!-- if RuntimeComponents is empty, we're ok because we included all the stubs already -->
      <!-- if RuntimeComponents is not empty, remove stubs for the required components and include the actual component library -->
```

Reply:
Fixed.